pkg/abaputils/abaputils.go

```diff
@@ -0,0 +1,291 @@
+	"regexp"
+	"strings"
+
+	"github.com/SAP/jenkins-library/pkg/cloudfoundry"
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/pkg/errors"
+)
+
+// ReadServiceKeyAbapEnvironment from Cloud Foundry and returns it.
+// Depending on user/developer requirements if he wants to perform further Cloud Foundry actions
+// the cfLogoutOption parameters gives the option to logout after reading ABAP communication arrangement or not.
+func ReadServiceKeyAbapEnvironment(options AbapEnvironmentOptions, c command.ExecRunner, cfLogoutOption bool) (AbapServiceKey, error) {
+
+	var abapServiceKey AbapServiceKey
+	var serviceKeyBytes bytes.Buffer
+	var err error
+
+	//Logging into Cloud Foundry
+	config := cloudfoundry.LoginOptions{
+		CfAPIEndpoint: options.CfAPIEndpoint,
+		CfOrg:         options.CfOrg,
+		CfSpace:       options.CfSpace,
+		Username:      options.Username,
+		Password:      options.Password,
+	}
+
+	err = cloudfoundry.Login(config, c)
+
+	c.Stdout(&serviceKeyBytes)
+	if err == nil {
+		// Reading Service Key
+		log.Entry().WithField("cfServiceInstance", options.CfServiceInstance).WithField("cfServiceKey", options.CfServiceKeyName).Info("Read service key for service instance")
+
+		cfReadServiceKeyScript := []string{"service-key", options.CfServiceInstance, options.CfServiceKeyName}
+
+		err = c.RunExecutable("cf", cfReadServiceKeyScript...)
+	}
+	if err == nil {
+		var serviceKeyJSON string
+
+		if len(serviceKeyBytes.String()) > 0 {
+			var lines []string = strings.Split(serviceKeyBytes.String(), "\n")
+			serviceKeyJSON = strings.Join(lines[2:], "")
+		}
+
+		json.Unmarshal([]byte(serviceKeyJSON), &abapServiceKey)
+		if abapServiceKey == (AbapServiceKey{}) {
+			return abapServiceKey, errors.New("Parsing the service key failed")
+		}
+
+		log.Entry().Info("Service Key read successfully")
+	}
+	if err != nil {
+		if cfLogoutOption == true {
+			var logoutErr error
+			logoutErr = cloudfoundry.Logout()
+			if logoutErr != nil {
+				return abapServiceKey, fmt.Errorf("Failed to Logout of Cloud Foundry: %w", err)
+			}
+		}
+		return abapServiceKey, fmt.Errorf("Reading Service Key failed: %w", err)
+	}
+
+	// Logging out of CF
+	if cfLogoutOption == true {
+		var logoutErr error
+		logoutErr = cloudfoundry.Logout()
+		if logoutErr != nil {
+			return abapServiceKey, fmt.Errorf("Failed to Logout of Cloud Foundry: %w", err)
+		}
+	}
+	return abapServiceKey, nil
+}
+
+// GetAbapCommunicationArrangementInfo function fetches the communcation arrangement information in SAP CP ABAP Environment
+// If no oData service URL is set, the MANAGE_GIT_REPOSITORY OData service will be used
+func GetAbapCommunicationArrangementInfo(options AbapEnvironmentOptions, c command.ExecRunner, oDataURL string, cfLogoutOption bool) (ConnectionDetailsHTTP, error) {
+
+	var connectionDetails ConnectionDetailsHTTP
+	var error error
+
+	if oDataURL == "" {
+		oDataURL = "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Pull"
+	}
```

Comment:
Personally, I would not set MANAGE_GIT_REPOSITORY as default. You could just leave it empty.

Reply:
No problem, I will fix this